tests/results/result_actions/ApplyPatchActionTest.py

```diff
@@ -91,6 +92,53 @@ def test_apply_orig_option(self):
 
             self.assertEqual(file_dict, expected_file_dict)
 
+    def test_apply_rename(self):
+        uut = ApplyPatchAction()
+        with make_temp() as f_a:
+            file_dict = {f_a: ["1\n", "2\n", "3\n"]}
+            expected_file_dict = {f_a: ["1\n", "2_changed\n", "3_changed\n"]}
+            file_diff_dict = {}
+            diff = Diff(file_dict[f_a], rename=f_a+".renamed")
+            diff.change_line(3, "3\n", "3_changed\n")
+            uut.apply(Result("origin", "msg", diffs={f_a: diff}),
+                      file_dict,
+                      file_diff_dict)
+            self.assertTrue(isfile(f_a+".orig"))
+            self.assertTrue(isfile(f_a+".renamed"))
+
+            diff = Diff(file_dict[f_a])
+            diff.change_line(2, "2\n", "2_changed\n")
+            uut.apply(Result("origin", "msg", diffs={f_a: diff}),
+                      file_dict,
+                      file_diff_dict)
+            self.assertTrue(isfile(f_a+".renamed.orig"))
+
+            for filename in file_diff_dict:
+                file_dict[filename] = file_diff_dict[filename].modified
```

Comment:
could you check if the contents of the actual file are good? Also shouldn't the filename in the expected dict be `f_a + '.renamed'`?